docs/user-guide/src/quick-start.md

```diff
@@ -0,0 +1,642 @@
+  disableNameSuffixHash: true
+# Create secrets with the credentials for accessing Ironic and Inspector.
+secretGenerator:
+  - name: ironic-credentials
+    files:
+      - username=ironic-username
+      - password=ironic-password
+  - name: ironic-inspector-credentials
+    files:
+      - username=ironic-inspector-username
+      - password=ironic-inspector-password
+```
+
+At this point, you should have a folder structure like this:
+
+```text
+bmo/
+├── ironic-inspector-password
+├── ironic-inspector-username
+├── ironic-password
+├── ironic-username
+├── ironic.env
+└── kustomization.yaml
+```
+
+You can check that the kustomization works and inspect the resulting manifest by
+running this:
+
+```bash
+kubectl create -k bmo --dry-run=client -o yaml
+```
+
+When you are happy with the output, apply it in the cluster:
+
+```bash
+kubectl apply -k bmo
+```
+
+## Deployment summary
+
+You are not expected to go through all the above steps each time you want to
+deploy Metal3. Store the configuration and reuse it the next time.
+
+Here is a summary of the deploy steps when all configuration is already in
+place.
+
+1. Create the management cluster.
+
+   ```bash
+   kind create cluster --config kind.yaml
+   ```
+
+2. Deploy cert-manager.
+
+   ```bash
+   kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.14.3/cert-manager.yaml
+   ```
+
+3. Start the DHCP server.
+
+   ```bash
+   docker run --name dnsmasq --rm -d --net=host --privileged --user 997:994 \
+     --env-file dnsmasq.env --entrypoint /bin/rundnsmasq \
+     quay.io/metal3-io/ironic
+   ```
+
+4. Start the image server.
+
+   ```bash
+   docker run --name image-server --rm -d -p 80:8080 \
+     -v "$(pwd)/disk-images:/usr/share/nginx/html" nginxinc/nginx-unprivileged
+   ```
+
+5. Deploy Ironic.
+
+   ```bash
+   kubectl apply -k ironic
+   ```
+
+6. Deploy Baremetal Operator.
+
+   ```bash
+   kubectl apply -k bmo
+   ```
+
+## Create BareMetalHosts
+
+Now that we have Baremetal operator deployed, let's put it to use by creating
+BareMetalHosts (BMHs) to represent our servers. You will need the protocol and
+IPs of the BMCs, as well as credentials for accessing them, and the servers MAC
+addresses.
+
+Create one secret for each BareMetalHost, containing the credentials for
+accessing its BMC. Here is an example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bml-01
+type: Opaque
+stringData:
+  username: replaceme
+  password: replaceme
+```
+
+Then continue by creating the BareMetalHost manifest. You can put it in the same
+file as the secret if you want. Just remember to separate the two resources with
+one line containing `---`.
+
+Here is an example of a BareMetalHost referencing the secret above with MAC
+address and BMC address matching our `bml-01` server.
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: bml-01
+spec:
+  online: true
+  bootMACAddress: 80:c1:6e:7a:e8:10
+  bootMode: legacy
```

Comment:
Is it really required in your case? We no longer recommend using legacy boot by default. At the very least, please explain this choice so that the users don't blindly copy it on newer hardware.

Reply:
Sadly it is :slightly_frowning_face: 
I actually tried first with UEFI but it was not supported on the hardware. I will add a comment about it though